lib/models/watcher.js

```diff
@@ -16,15 +18,88 @@ class Watcher extends CoreObject {
     logger.info('initialize %o', options);
 
     this.serving = _options.serving;
-    this.watcher = this.watcher || this.constructWatcher(options);
+
+    if (isExperimentEnabled('BROCCOLI_WATCHER')) {
+      this.watcher = this.watcher || this._constructBroccoliWatcher(options);
+      this.watcher.on('change', this.didChange.bind(this));
+      this.watcher.on('buildSuccess', this.didBuild.bind(this));
+      this.watcher.start();
+      this._setupBuild(this.watcher.currentBuild);
+    } else {
+      this.watcher = this.watcher || this._constructWatcher(options);
+      this.watcher.on('change', this.didBuild.bind(this));
+      this.build = this.watche;
+    }
 
     this.watcher.on('error', this.didError.bind(this));
-    this.watcher.on('change', this.didChange.bind(this));
     this.serveURL = serveURL;
   }
 
-  constructWatcher(options) {
-    return new (require('ember-cli-broccoli-sane-watcher'))(this.builder, options);
+  _constructWatcher(options) {
+    const watcher = new (require('ember-cli-broccoli-sane-watcher'))(this.builder, options);
+    this.build = watcher;
```

Comment:
`_constructWatcher` having side-affects, by placing `watcher` on `this.build`, is surprising. It should likely be symmetrical with where else `this.build` is set.